src/jitlayers.cpp

```diff
@@ -500,7 +498,11 @@ OptimizerResultT JuliaOJIT::OptimizerT::operator()(orc::ThreadSafeModule TSM, or
 
         JL_TIMING(LLVM_OPT);
 
-        PM.run(M);
+        {
+            //Lock around our pass manager
+            std::lock_guard<std::mutex> lock(this->mutex);
```

Comment:
priority inversion? Can you add these locks to doc/src/devdocs/locks.md

Reply:
Would this be considered a level 1 lock since it never holds another lock within its scope?

Reply:
Yeah, if it does not need any others, which seems might be the case? That means our LLVM passes must not use any Julia state (global lookups, etc), but that seems probably true anyways.

Reply:
This probably doesn't affect us right now, since all of our work is done under the codegen lock anyways, but I wonder if the targetmachine used in llvm-cpufeatures will bite us here later then.